osu.Game.Rulesets.Osu/Mods/OsuModRandom.cs

```diff
@@ -0,0 +1,364 @@
+
+            // Move hit objects back into the playfield if they are outside of it,
+            // which would sometimes happen during big jumps otherwise.
+            if (position.X < 0)
+                position.X = 0;
+            else if (position.X > OsuPlayfield.BASE_SIZE.X)
+                position.X = OsuPlayfield.BASE_SIZE.X;
+
+            if (position.Y < 0)
+                position.Y = 0;
+            else if (position.Y > OsuPlayfield.BASE_SIZE.Y)
+                position.Y = OsuPlayfield.BASE_SIZE.Y;
+
+            return position;
+        }
+
+        /// <summary>
+        /// Determines the position of the current hit object relative to the previous one.
+        /// </summary>
+        /// <returns>The position of the current hit object relative to the previous one</returns>
+        private Vector2 getRotatedVector(Vector2 prevPosChanged, Vector2 posRelativeToPrev)
+        {
+            var relativeRotationDistance = 0f;
+            var playfieldMiddle = Vector2.Divide(OsuPlayfield.BASE_SIZE, 2);
+
+            if (prevPosChanged.X < playfieldMiddle.X)
+            {
+                relativeRotationDistance = Math.Max(
+                    (border_distance_x - prevPosChanged.X) / border_distance_x,
+                    relativeRotationDistance
+                );
+            }
+            else
+            {
+                relativeRotationDistance = Math.Max(
+                    (prevPosChanged.X - (OsuPlayfield.BASE_SIZE.X - border_distance_x)) / border_distance_x,
+                    relativeRotationDistance
+                );
+            }
+
+            if (prevPosChanged.Y < playfieldMiddle.Y)
+            {
+                relativeRotationDistance = Math.Max(
+                    (border_distance_y - prevPosChanged.Y) / border_distance_y,
+                    relativeRotationDistance
+                );
+            }
+            else
+            {
+                relativeRotationDistance = Math.Max(
+                    (prevPosChanged.Y - (OsuPlayfield.BASE_SIZE.Y - border_distance_y)) / border_distance_y,
+                    relativeRotationDistance
+                );
+            }
+
+            return rotateVectorTowardsVector(
+                posRelativeToPrev,
+                Vector2.Subtract(playfieldMiddle, prevPosChanged),
+                relativeRotationDistance
+            );
+        }
+
+        /// <summary>
+        /// Rotates vector "initial" towards vector "destinantion"
+        /// </summary>
+        /// <param name="initial">Vector to rotate to "destination"</param>
+        /// <param name="destination">Vector "initial" should be rotated to</param>
+        /// <param name="relativeDistance">The angle the vector should be rotated relative to the difference between the angles of the the two vectors.</param>
+        /// <returns>Resulting vector</returns>
+        private Vector2 rotateVectorTowardsVector(Vector2 initial, Vector2 destination, float relativeDistance)
```

Comment:
I don't really understand what this is doing. Do you mind explaining? As far as I can tell this doesn't really "rotate vector towards vector" rather than just perturb the angles between them within a 180 degree interval.

Reply:
Let's say `initial` has an angle of 45° and `destination` an angle of 180°.
If `relativeDistance` is set to 0.5, it will return `initial` rotated half way to `destination` (to 112.5°).
If `relativeDistance` is set to 1 it will be rotated all the way to `destination` and if it's 0 nothing will happen.

I made visualisation here: https://www.geogebra.org/calculator/zd7rbe3x
v.initial, v.destination and relativeDistance would be the parameters and v.result would be the returned `Vector2`.
(You can also change relativeDistance and move the points the vectors are pointing to to see how v.result changes.)